CSLabs.Api/Controllers/SystemMessagesController.cs

```diff
@@ -0,0 +1,51 @@
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using CSLabs.Api.Models;
+using CSLabs.Api.Models.ModuleModels;
+using CSLabs.Api.RequestModels;
+using CSLabs.Api.Util;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+
+namespace CSLabs.Api.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    [Authorize]
+
+    public class SystemMessagesController : BaseController
+    {
+        public SystemMessagesController(BaseControllerDependencies dependencies) : base(dependencies)
+        {
+
+
+            [HttpGet("{id}")] 
+            public async Task<IActionResult> GetSystemMessageID(int id)
+            {
+                return; 
+            }
+
+            [HttpPost] //Create
+            public async Task<IActionResult> UpsertSystemMessage()
```

Comment:
Upsert stands for update or insert. So either have an upsert method or an update and create method but not both. (you have both upsert and UpdateSystemMessage methods)